nyc.config.js

```diff
@@ -1,7 +1,7 @@
 const config = require('./jest.config');
 
 const removeRootDirectory = pathArray => {
-  return pathArray.map(path => path.slice(10));
+  return pathArray.map(path => path.replace('(.*)', '**').slice(10));
```

Comment:
Did this do anything? We still want to read `collectCoverageFrom`, right?

Reply:
Oh my bad. I totally misunderstood the part about using ``collectCoverageFrom``.

I believe with this modification, ``components/UpdateProfileForm/steps/index.js`` is properly ignored during Cypress testing.

What I am really targeting with this modification are ``'<rootDir>/common/(.*)/index.js'`` and ``'<rootDir>/components/(.*)/index.js'`` from ``coveragePathIgnorePatterns`` and basically replacing ``(.*)`` with ``**``. Any other paths in ``collectCoverageFrom`` and  ``coveragePathIgnorePatterns`` will not be affected since those two are the only ones with that pattern.

But we can definitely use ``collectCoverageFrom`` as we initially discussed. Let me know what you think!

Reply:
I see now. I guess I'm just surprised we didn't do:

```js
const removeRootDirFromPaths = (pathsArray) => {
  return pathsArray.map(path => path.replace('<rootDir>/', ''));
}

// ...

  include: ['pages/**/*.js'],
  exclude: [
    ...removeRootDirFromPaths(config.collectCoverageFrom), 
    ...removeRootDirFromPaths(config.coveragePathIgnorePatterns)),
  ]
```

Reply:
Or... do we even need the `exclude` array?

Reply:
Correct. I believe we won't need ``exclude`` array if we only included ``pages/**/*.js``

But, I think we need to at least exclude ``pages/api/__coverage.js__`` there.

And I believe the overall test coverage result will benefit if we include the ``config.collectCoverageFrom``, at least I am certain that including ``common/**/*.js`` and ``decorators/**/*.js`` has increased the total percentage. 

So, I think it really comes down to what we include in the coverage and depending on that, we can decide what to exclude from the result.

Reply:
Wouldn't the `jest` config capture that though, and then the reports get merged.

Reply:
Yep. Jest definitely does.

But I am pretty sure that test coverage for certain files such as ``decorators/withAuthSync/withAuthSync.js`` went up after the merge when those were included in the Cypress test. 

To be sure, I think we would be able to check if there is any difference in the result on CodeClimate if I make a new commit with the modified config as below.

```
include: ['pages/**/*.js'],
exclude: ['pages/api/__coverage__.js']
```

Reply:
Yes, coverage goes up because everything gets called again. In reality, `withAuthSync` can have it's own unit/integration test with Jest, but it doesn't have one. Although, it would be hard to test without the context of full-blown routing...

Alright, let's go with this - final answer:

```
include: ['pages/**/*.js', 'decorators/withAuthSync/withAuthSync.js'],
exclude: ['pages/api/__coverage__.js']
```